### DISCUSSION TITLE
Odoo-ORM Isolation from Business Logic

### DISCUSSION THREAD

**Discussion by Rodesc** [OCA/odoo-community.org#152] [Ideas]
Hello Everyone,

I started working on the isolation of the Odoo-ORM as I want to use the ORM to build a completely unrelated software on top of it. My goal is to keep only the ORM stripped from all business logic.

Does anyone want to help with this ? Maybe it should be included in the OCA organization, no idea how to submit a new repo.

Curious to hear your thoughts.

https://github.com/Rodesc/odoo-orm

**Comment by jguenat** [comment]
Hello,

Maybe you already though about that but if your app depends only on base and no other app is installed you are kind of stripped from any Odoo business logic.

**Comment by HekkiMelody** [comment]
I'd be curious to know what you see as the strengths of the Odoo ORM by itself, such that you want to use it without the rest of Odoo, especially when compared to other popular ORMs for python, such as the Django ORM, or SQLAlchemy.

If you're already going to such lengths, the Django ORM in particular seems to me to be a clear upgrade over the Odoo ORM in every way. Maybe there are aspects I haven't considered, of course!

**Reply by ap-wtioit** [reply]
there was a talk for this on Odoo Experience 2022

https://www.odoo.com/event/odoo-experience-2022-2190/track/non-erp-odoo-using-odoo-core-for-developing-non-business-applications-4907

**Reply by HekkiMelody** [reply]
Thank you for the link, I'll give it a watch

**Reply by HekkiMelody** [reply]
I disagree with a lot of what the talk says, personally. Like with anything Odoo, it's great as long as you need exactly what it offers, and it's hell the moment you need to make a customization. It's also a big monolith, so you can't mix and match pieces as needed, you have to take it all or leave it all.

**Reply by SirAionTech** [reply]
Another comparison from last year (I think they do this kind of talk every year) https://www.odoo.com/it_IT/event/odoo-experience-2023-3735/track/why-the-odoo-orm-is-not-like-other-orms-5811, clearly the talks **from Odoo** say that Odoo is better than anything else :wink:, I'd be more interested in some video from a third party.

I don't know other ORMs well enough, but is it really that easy to create computed fields or multi-DB support in other ORMs?

**Comment by moyoseph** [comment]
hey, are you still working on this or is it abandoned?
